core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponentSpec.scala

```diff
@@ -652,6 +651,21 @@ class EntityComponentSpec extends TestDriverComponentWithFlatSpecAndMatchers wit
           assertResult(expectedEntities) {
             runAndWait(entityQuery.listActiveEntities(cloneContext)).toSet
           }
+          val destShardId = determineShard(cloneContext.workspaceIdAsUUID)
+          val destWsId = cloneContext.workspaceIdAsUUID
+          val badEntityReferenceCount = sql"""select count(*) from ENTITY e join ENTITY_ATTRIBUTE_#$destShardId ea on e.id = ea.owner_id
+                        join ENTITY e_ref on ea.value_entity_ref = e_ref.id
+                        where ea.value_entity_ref is not null and e_ref.workspace_id != $destWsId and e.workspace_id = $destWsId"""
+          assertResult(0, "cloned entity references should only point to entities within the same workspace") {
+            runAndWait(badEntityReferenceCount.as[Int].head)
+          }
+
+          val expectedEntityReferenceCount = sql"""select count(*) from ENTITY e join ENTITY_ATTRIBUTE_#$destShardId ea on e.id = ea.owner_id
+                        join ENTITY e_ref on ea.value_entity_ref = e_ref.id
+                        where ea.value_entity_ref is not null and e_ref.workspace_id = $destWsId and e.workspace_id = $destWsId"""
+          assertResult(3, "cloned entity references should only point to entities within the same workspace") {
```

Comment:
nit: I think it would be helpful if this error message was slightly different to highlight the success case. How about something like "all entity references should be cloned to the destination workspace. The entity reference count in the destination workspace does not match the initial count from the source workspace". Err my error is a little wordy, just the "all entity references should be clone to the destination workspace" would be good